service/topology_coordinator.cc

```diff
@@ -756,6 +760,83 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
         case global_topology_request::cleanup:
             co_await start_cleanup_on_dirty_nodes(std::move(guard), true);
             break;
+        case global_topology_request::keyspace_rf_change: {
```

Comment:
IMO it would be nice to move the code below into a separate function.

Reply:
The refactor is not necessary for the code to work, and because of the limited time I thought I'm going to clean the code in a follow-up.

Reply:
Issue created https://github.com/scylladb/scylladb/issues/19126